module_configuration.py

```diff
@@ -28,28 +31,74 @@ def install_client_pkg(client_path):
         print(f"pip install {client_path}")
 
 
-def configure_server_module():
-    print("Not Implemented")
+def get_server_modules(cfg):
+    mod_list = []
+    with open(cfg) as f:
+        for line in f.readlines():
+            m = line.split("#")[0].split("/")[-1].strip()
+            mod_list += [m] if m else []
+    return mod_list
+
+
+def get_chpl_files(mod_path):
+    path = mod_path + "/server/*.chpl"
+    return glob.glob(path)
+
+
+def configure_server_module(mod_path, ak_loc):
+    if not ak_loc:
+        raise RuntimeError("--ak option is required when configuring a module with server components.")
+    mod_cfg = mod_path + "/server/ServerModules.cfg"
+    ak_cfg = ak_loc + "/ServerModules.cfg"
+    if not os.path.exists(mod_cfg):
+        raise RuntimeError(f"Could not locate module ServerModules.cfg: {mod_cfg}")
+    if not os.path.exists(ak_cfg):
+        raise RuntimeError(f"Could not locate module ServerModules.cfg: {ak_cfg}")
+
+    # get all of the modules listed in the .cfg file
+    mods = get_server_modules(mod_cfg)
 
+    # Get chpl files that will be added to arkouda
+    c_files = get_chpl_files(mod_path)
 
-def run(mod_path):
-    client_path = mod_path+"/client"
-    server_path = mod_path+"/server"
-    if os.path.exists(client_path):
+    # Generate commands - these can be piped into a shell for execution
+    # 1 make copy of the Arkouda ServerModules.cfg
+    tmp_cfg = f"{mod_path}/TmpServerModules.cfg.{int(time.time())}"
+    print(f"cp {ak_cfg} {tmp_cfg}")
+
+    #2 append our modules
+    for c in mods:
+        print(f"echo {c} >> {tmp_cfg}")
+
+    #3 generate make command with vars
+    ak_srv_user_mods = '"' + " ".join(c_files) + '"' # setup our ARKOUDA_SERVER_USER_MODULES
+    print(f"ARKOUDA_SERVER_USER_MODULES={ak_srv_user_mods} ARKOUDA_CONFIG_FILE={tmp_cfg} "
+          f"ARKOUDA_SKIP_CHECK_DEPS=true make -C {ak_loc}")
+
+def run(mod_path, ak_loc):
+    mod_path = mod_path.rstrip("/")  # remove trailing slash
+    ak_loc = ak_loc.rstrip("/")  # remove trailing slash
+
+    client_path = mod_path + "/client"
+    server_path = mod_path + "/server"
+
+    if os.path.exists(mod_path):
         install_client_pkg(client_path)
-        if os.path.isdir(mod_path+"/server"):
-            configure_server_module()
+        if os.path.exists(server_path) and os.path.exists(ak_loc):
+            if os.path.exists(ak_loc):
```

Comment:
is this check redundant?

Reply:
updated the if so that we first check if server path exists because we only need the arkouda location if a server module is present.